hazelcast/src/main/java/com/hazelcast/nio/serialization/genericrecord/GenericRecord.java

```diff
@@ -511,6 +511,10 @@ public interface GenericRecord {
     OffsetDateTime[] getArrayOfTimestampWithTimezone(@Nonnull String fieldName);
 
     /**
+     * For {@link com.hazelcast.config.CompactSerializationConfig Compact}
+     * objects, it is not allowed write an array containing different item
+     * types or a {@link HazelcastSerializationException} will be thrown.
+     *
```

Comment:
🤔 Why put this warning to getArrayOfGenericRecord? What about putting to GenericRecordBuilder.setArrayOfGenericRecord instead?

Reply:
Makes sense, I moved it to there in https://github.com/hazelcast/hazelcast/pull/21958/commits/e7ef1b4c187571aedd60f926537fbfbdfd51df70